packages/wonder-blocks-tooltip/components/tooltip-arrow.js

```diff
@@ -0,0 +1,232 @@
+            case "top":
+                return {
+                    points: [
+                        "0,0",
+                        `${arrowWidth / 2},${arrowHeight}`,
+                        `${arrowWidth},0`,
+                    ],
+                    height: arrowHeight,
+                    width: arrowWidth,
+                };
+
+            case "right":
+                return {
+                    points: [
+                        `${arrowHeight},0`,
+                        `0,${arrowWidth / 2}`,
+                        `${arrowHeight},${arrowWidth}`,
+                    ],
+                    width: arrowHeight,
+                    height: arrowWidth,
+                };
+
+            case "bottom":
+                return {
+                    points: [
+                        `0, ${arrowHeight}`,
+                        `${arrowWidth / 2},0`,
+                        `${arrowWidth},${arrowHeight}`,
+                    ],
+                    width: arrowWidth,
+                    height: arrowHeight,
+                };
+
+            case "left":
+                return {
+                    points: [
+                        `0,0`,
+                        `${arrowHeight},${arrowWidth / 2}`,
+                        `0,${arrowWidth}`,
+                    ],
+                    width: arrowHeight,
+                    height: arrowWidth,
+                };
+
+            default:
+                throw new Error(`Unknown placement: ${placement}`);
+        }
+    }
+
+    _renderArrow() {
+        const {placement} = this.props;
+        const {
+            points,
+            height,
+            width,
+        } = this._calculateDimensionsFromPlacement();
+
+        // TODO(somewhatabstract): Use unique IDs!
+        const dropShadowFilterId = `tooltip-dropshadow-${placement}`;
+
+        return (
+            <svg
+                className={css(styles.arrow, styles[`arrow-${placement}`])}
+                width={width}
+                height={height}
+            >
+                {/* Create an SVG filter that applies a blur to an element.
+                  * We'll apply it to a dark shape outlining the tooltip, which
+                  * will produce the overall effect of a drop-shadow.
+                  *
+                  * Also, scope its ID by side, so that tooltips with other
+                  * "side" values don't end up using the wrong filter from
+                  * elsewhere in the document. (The `height` value depends on
+                  * which way the arrow is turned!)
+                  *
+                  * In general, it's not *great* that multiple tooltips of the
+                  * same side will yield ID conflicts... but at least they'll
+                  * be the same filter, so it doesn't matter which one we
+                  * reference, so long as it's for the correct side. */}
+                <filter id={dropShadowFilterId} height="150%">
```

Comment:
Every time I look at this filter my eyes glaze over.  How can CSS box-shadow be so simple and this be so complicated!?

Reply:
I know. I hate it. I can't even get it exactly how I want. I will..I will fight until I do, but still. Ugh.